sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3511,6 +3511,17 @@ ur_result_t ExecCGCommand::enqueueImpQueue() {
     const AdapterPtr &Adapter = MQueue->getAdapter();
     if (MEvent != nullptr)
       MEvent->setHostEnqueueTime();
+    // Honor explicit dependencies set with depends_on.
+    // Important for cross queue dependencies.
+    // Intentionally do not sort out events on the same queue (no sense).
+    // Sync of events wait to be done by the following barrier
+    if (RawEvents.size()) {
+      auto Result = Adapter->call_nocheck<UrApiKind::urEnqueueEventsWait>(
+          MQueue->getHandleRef(), RawEvents.size(),
+          RawEvents.size() ? &RawEvents[0] : nullptr, nullptr);
+      if (Result != UR_RESULT_SUCCESS)
+        return Result;
+    }
```

Comment:
Maybe I am missing something, but I believe we can just pass RawEvents as waitlist to `urEnqueueEventsWaitWithBarrierExt` call. It seems like it would be better in terms of performance - less UR/L0 calls, less objects created etc.

Reply:
for barrier without wait list events provided - no, it will be not correct. barrier w/o events waits for all previous submissions to the same queue. barrier w/ events waits only for the specified list of events.

Reply:
Indeed, thanks.